packages/ui/src/FileCard/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { ReactNode } from "react";
+
+import { IFile } from "../FilesTable";
+
+export type Messages = {
+  downloadCountLabel?: string;
+  downloadButtonLabel?: string;
+  lastDownloadedAtHeader?: string;
+  uploadedbyHeader?: string;
+  uploadedAtHeader?: string;
+  viewButtonLabel?: string;
+};
+
+type FileCardType = {
+  file: IFile;
+  messages?: Messages;
+  onArchive?: (arguments_: IFile) => void;
+  onDelete?: (arguments_: IFile) => void;
+  onDownload?: (arguments_: IFile) => void;
+  onShare?: (arguments_: IFile) => void;
+  onView?: (arguments_: IFile) => void;
+  renderThumbnail?: (arguments_: IFile) => ReactNode;
+  showThumbnail?: boolean;
+  showDescription?: boolean;
+};
+
+export const FileCard = ({
+  file,
+  onArchive,
+  onDelete,
+  onDownload,
+  onShare,
+  onView,
+  messages,
+  renderThumbnail,
+  showThumbnail = true,
+  showDescription = true,
+}: FileCardType) => {
+  const checkUploadedByData = (data: any) => {
+    if (!data.uploadedBy) {
+      return <code>&#8212;</code>;
+    }
+
+    if (data.uploadedBy.givenName || data.uploadedBy.lastName) {
+      return `${data.uploadedBy.givenName || ""} ${
+        data.uploadedBy.lastName || ""
+      }`;
+    }
+
+    return data.uploadedBy.email;
+  };
+
+  const renderIconThumbnail = () => {
+    if (showThumbnail) {
+      if (!renderThumbnail) {
+        return (
+          <div className="file-icon">
+            <i className="pi pi-file-pdf"></i>;
+          </div>
+        );
+      }
+      return <div className="file-icon">{renderThumbnail?.(file)}</div>;
+    }
+    return null;
+  };
+
+  return (
+    <Card className="file-card">
+      <div className="file-card-wrapper">
+        {renderIconThumbnail()}
+        <div className="file-details-wrapper">
+          <div className="file-details">
+            <span>
+              <span className="file-name">{file.filename}</span>
+              <span className="file-size">
+                {file.size && `(${file?.size})`}
+              </span>
```

Comment:
This part should be configurable. Dev must be able to choose whether or not to display file size.